src/client/kube_store/index.ts

```diff
@@ -233,6 +48,47 @@ export default class KubeStore {
     return this.instance || (this.instance = new this(''));
   }
 
+  public patchResource(resource: KubeResource, name: string, patch: object): object {
+    let source: object = {};
+    let key: string = '';
+    const myPatch: object = {'foo':'bar'};
+    if (resource instanceof NamespacedResource) {
```

Comment:
As we discussed, later when we know this function is working, we can explore refactoring it so that determining the key is in a different function which can be reused.

Reply:
Looks like you are ignoring this entire block and not using the 'result'

The 'result' is the important part of how we do the lookup of mocked data